src/OptionalPackagesInstaller.php

```diff
@@ -318,7 +318,7 @@ private function runInstaller(RootPackageInterface $package, Collection $package
         );
 
         $installer->setDevMode(true);
-        $installer->setUpdate();
+        $installer->setUpdate(false);
```

Comment:
This is incorrect.

In the v1 releases of Composer, that method provided a default value for `$update` of `true`. As such, passing `false` is the opposite of what we were doing previously.

I'm going to push a change to this to your branch, as well as updates to the tests so that they pass.